### PR TITLE
Updated pre-commit so that it can run locally on every commit

### PR DESCRIPTION
**Pull request by nikhil25803**
Issue addressed - #2419 

Screenshot :
image

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -76,6 +76,7 @@ Markdown = "^3.6"
 faiss-cpu = "^1.8.0"
 openai = "^1.35.13"
 psutil = "^5.9.8"
+pre-commit = "^3.7.1"
```
Comment (arkid15r) [inline pyproject.toml:79]: This should go under `[tool.poetry.group.dev.dependencies]`.

**Review by arkid15r — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (nikhil25803) [reply pyproject.toml]: Done

**Review by arkid15r — COMMENTED** [review]
👍

**Review by arkid15r — DISMISSED** [review]

**Comment by nikhil25803** [comment]
I am getting this error on running test cases.
image

Related to `poetry install`.
Error message - `pyproject.toml changed significantly since poetry.lock was last generated. Run `poetry lock [--no-update]` to fix the lock file.`

**Comment by nikhil25803** [comment]
@arkid15r 
Do I need to make changes in `ci-cd.yml` file to resolve this issue?
image


Adding `run: poetry lock` before `run: poetry install`/

**Comment by arkid15r** [comment]
> @arkid15r Do I need to make changes in `ci-cd.yml` file to resolve this issue? image
> 
> Adding `run: poetry lock` before `run: poetry install`/

No, just update your poetry.lock locally

**Comment by nikhil25803** [comment]
I tried @arkid15r , but still getting the same error
```
pyproject.toml changed significantly since poetry.lock was last generated. Run `poetry lock [--no-update]` to fix the lock file.
Error: Process completed with exit code 1.
```

**Review by arkid15r — APPROVED** [review]
LGTM

**Pull request merged**